registry/modules/auth-basic/controllers/authBasic.ts

```diff
@@ -0,0 +1,61 @@
+import argon from "argon2";
+
+import { BasicAuthSchema } from "@/schemaValidators/auth-basic.interface.js";
+
+import { accessTokenManager } from "@/modules/auth-basic/utils/jwt/tokenManager.js";
+import {
+  usernameNotAvailable,
+  invalidLoginCredentials,
+} from "@/modules/auth-basic/utils/errors/auth.js";
+
+import { UserRepository } from "@/repositories/user.interface.js";
+
+export const createAuthBasicController = (userRepo: UserRepository) => {
```

Comment:
Shouldnt `accessTokenManager` also be passed as a argument to this factory?

Reply:
It can be useful if we want to write test with different `accessTokenManager` implementations. Not sure how necessary that is atm, that being said, could this be breaking "best practices" when building the modules with Dependency Injections?